src/index.js

```diff
@@ -46,47 +44,58 @@ module.exports = {
     'id-length': ['error', {
       exceptions: ['_', 'e', 'i']
     }],
+    'id-match': ['error', '^[_a-zA-Z][a-zA-Z0-9]*$|^[A-Z][A-Z]+_[A-Z][A-Z]+$'],
```

Comment:
No, I've tested some variants, but it surely can be improved.

Your suggestion means we accept something such as:

```js
const A1_2 = 'foo';
const A1_B = 'bar';
```

which might be a bit weird. What do you think?